irmago_test.go

```diff
@@ -985,6 +985,375 @@ func TestWizardFromScheme(t *testing.T) {
+			}}},
+		},
+	}
+
+	err := wizard.Validate(conf)
+	require.Equal(t, "wizard with wizard ID testwizard too complex", err.Error())
+}
+
+func TestIssueWizardItemValidation(t *testing.T) {
+	dependencies := CredentialDependencies{
+		{
+			{credid("a.a.a"), credid("a.a.b")},
+			{credid("a.b.a")},
+		},
+		{
+			{credid("b.a.a")},
+			{credid("b.b.a"), credid("b.b.b")},
+		},
+	}
+
+	conf := &Configuration{CredentialTypes: map[CredentialTypeIdentifier]*CredentialType{}}
+	for _, discon := range dependencies {
+		for _, con := range discon {
+			for _, cred := range con {
+				conf.CredentialTypes[cred] = credtype(cred.String())
+			}
+		}
+	}
+
+	credTypId := NewCredentialTypeIdentifier("a.a.a")
+	tester := IssueWizardItem{Type: IssueWizardItemTypeCredential, Credential: &credTypId, Text: &TranslatedString{"en": "text", "nl": "tekst"}}
+	schemeMan := SchemeManager{}
+	conf.SchemeManagers = map[SchemeManagerIdentifier]*SchemeManager{credTypId.SchemeManagerIdentifier(): &schemeMan}
+
+	credTypId2 := credid("a.b.a")
+	conf.CredentialTypes[credTypId].Dependencies = CredentialDependencies{
+		{
+			{credTypId2},
+		},
+	}
+
+	translation := trivialTranslation("Age limit")
+	conf.CredentialTypes[credTypId2].FAQSummary = &translation
+
+	require.Equal(t, nil, tester.validate(conf))
+}
+
+func TestIssueWizardFAQSummariesValidation(t *testing.T) {
+	dependencies := CredentialDependencies{
+		{
+			{credid("a.a.a"), credid("a.a.b")},
+			{credid("a.b.a")},
+		},
+		{
+			{credid("b.a.a")},
+			{credid("b.b.a"), credid("b.b.b")},
+		},
+	}
+
+	conf := &Configuration{CredentialTypes: map[CredentialTypeIdentifier]*CredentialType{}}
+	for _, discon := range dependencies {
+		for _, con := range discon {
+			for _, cred := range con {
+				conf.CredentialTypes[cred] = credtype(cred.String())
+			}
+		}
+	}
+
+	credTypId := NewCredentialTypeIdentifier("a.a.a")
+	tester := IssueWizardItem{Type: IssueWizardItemTypeCredential, Credential: &credTypId}
+	schemeMan := SchemeManager{}
+	conf.SchemeManagers = map[SchemeManagerIdentifier]*SchemeManager{credTypId.SchemeManagerIdentifier(): &schemeMan}
+
+	require.Equal(t, "FAQSummary missing for wizard item with credential type: a.a.a", tester.validate(conf).Error())
+
+	tester.Text = &TranslatedString{"en": "text"}
+	require.Equal(t, "Wizard item text field incomplete for item with credential type: a.a.a", tester.validate(conf).Error())
+
+	tester.Text = nil
+	conf.CredentialTypes[credTypId].FAQSummary = &TranslatedString{"en": "text"}
+	require.Equal(t, "FAQSummary missing for: a.a.a", tester.validate(conf).Error())
+
+	tester.Text = &TranslatedString{"en": "text", "nl": "tekst"}
+	conf.CredentialTypes[credTypId].Dependencies = CredentialDependencies{
+		{
+			{credid("a.b.a")},
+		},
+	}
+	require.Equal(t, "FAQSummary missing for last item in chain: a.a.a, a.b.a", tester.validate(conf).Error())
+}
+
+func TestCircularDependenciesValidation(t *testing.T) {
+	credTypeIdA := credid("scheme.issuer.a")
```

Comment:
When putting acronyms in variable names there is a mild preference for using all caps for the acronym, see e.g. https://github.com/golang/go/wiki/CodeReviewComments#initialisms. (This is, however, not consistently done across irmago).